frontend_vue/src/components/tokens/aws_infra/plan_generator/AssetCard.vue

```diff
@@ -107,40 +84,35 @@
 </template>
 
 <script lang="ts" setup>
-import { ref, computed, watch } from 'vue';
-import type { AssetDataType, S3ObjectType } from '../types';
+import { ref, computed } from 'vue';
 import getImageUrl from '@/utils/getImageUrl';
 import {
   ASSET_DATA_NAME,
   ASSET_LABEL,
   ASSET_WITH_ICON,
   AssetTypesEnum,
 } from '@/components/tokens/aws_infra/constants.ts';
+import type { AssetDataType } from '../types';
 
-type AssetDataTypeWithoutS3Object = Exclude<AssetDataType, S3ObjectType>;
+type VisibleAssetPropertiesType = [keyof AssetDataType, string | number];
```

Comment:
do we need this confusing type just to call `ASSET_DATA_NAME[props.assetType];`?